gloo/transport/ibverbs/device.cc

```diff
@@ -75,16 +75,31 @@ std::shared_ptr<::gloo::transport::Device> CreateDevice(
     }
     std::vector<std::string> names;
     for (auto i = 0; i < devices.size(); i++) {
+      GLOO_DEBUG(
+          "found candidate device ",
+          devices[i]->name,
+          " dev=",
+          devices[i]->dev_name);
       names.push_back(devices[i]->name);
     }
     std::sort(names.begin(), names.end());
     attr.name = names[0];
   }
 
+  attr.index = 3;
```

Comment:
N00b question, why this magic number? Is it possible that we can add a comment for it?

Reply:
Removed -- this is infra specific, should be passing it in via the attr and not hardcoding